tests/integration-tests/tests/dashboard_and_alarms/test_dashboard_and_alarms.py

```diff
@@ -0,0 +1,173 @@
+import datetime
+import math
+import time
+
+import boto3
+import pytest
+from assertpy import assert_that
+from botocore.exceptions import ClientError
+
+
+@pytest.mark.usefixtures("instance", "os", "scheduler")
+@pytest.mark.parametrize("dashboard_enabled, cw_log_enabled", [(True, True), (True, False), (False, False)])
+def test_dashboard_and_alarms(
+    dashboard_enabled,
+    cw_log_enabled,
+    region,
+    pcluster_config_reader,
+    clusters_factory,
+    test_datadir,
+):
+    cluster_config = pcluster_config_reader(
+        dashboard_enabled=str(dashboard_enabled).lower(),
+        cw_log_enabled=str(cw_log_enabled).lower(),
+    )
+    cluster = clusters_factory(cluster_config)
+    cw_client = boto3.client("cloudwatch", region_name=region)
+    headnode_instance_id = cluster.get_cluster_instance_ids(node_type="HeadNode")[0]
+    compute_instance_ids = cluster.get_cluster_instance_ids(node_type="Compute")
+
+    # test CWAgent metrics
+    # sleep for 2 minutes to ensure we can get metrics data
+    time.sleep(60 * 2)
+    start_timestamp, end_timestamp = _get_start_end_timestamp(minutes=2)
+    metrics_response_headnode = _get_metric_data(headnode_instance_id, cw_client, start_timestamp, end_timestamp)
+    mem_values = _get_metric_data_values(metrics_response_headnode, "mem")
+    disk_values = _get_metric_data_values(metrics_response_headnode, "disk")
+    assert_that(mem_values).is_not_empty()
+    assert_that(disk_values).is_not_empty()
+
+    if compute_instance_ids:
+        metrics_response_compute = _get_metric_data(compute_instance_ids[0], cw_client, start_timestamp, end_timestamp)
+        mem_values = _get_metric_data_values(metrics_response_compute, "mem")
+        disk_values = _get_metric_data_values(metrics_response_compute, "disk")
+        assert_that(mem_values).is_empty()
+        assert_that(disk_values).is_empty()
+
+    # test dashboard and alarms
+    dashboard_name = "{0}-{1}".format(cluster.cfn_name, region)
+    if dashboard_enabled:
+        # test dashboard
+        dashboard_response = cw_client.get_dashboard(DashboardName=dashboard_name)
+        assert_that(dashboard_response["DashboardName"]).is_equal_to(dashboard_name)
+        if cw_log_enabled:
+            assert_that(dashboard_response["DashboardBody"]).contains("Head Node Logs")
+        else:
+            assert_that(dashboard_response["DashboardBody"]).does_not_contain("Head Node Logs")
+
+        # test alarms
+        mem_alarm_name = f"{cluster.cfn_name}_MemAlarm_HeadNode"
+        disk_alarm_name = f"{cluster.cfn_name}_DiskAlarm_HeadNode"
+        alarm_response = cw_client.describe_alarms(AlarmNamePrefix=cluster.cfn_name)
+        mem_alarms = _get_alarm_records(alarm_response, mem_alarm_name)
+        disk_alarms = _get_alarm_records(alarm_response, disk_alarm_name)
+        _verify_alarms(mem_alarms, "mem_used_percent", headnode_instance_id)
+        _verify_alarms(disk_alarms, "disk_used_percent", headnode_instance_id)
```

Comment:
What about we wrap this into a separate function _test_alarms()?

Reply:
sure